frontend/src/pages/NewRunV2.tsx

```diff
@@ -72,11 +80,42 @@ function NewRunV2(props: NewRunV2Props) {
   const [isStartingNewRun, setIsStartingNewRun] = useState(false);
   const [errorMessage, setErrorMessage] = useState('');
   const [isParameterValid, setIsParameterValid] = useState(false);
+  const [clonedRuntimeConfig, setClonedRuntimeConfig] = useState<PipelineSpecRuntimeConfig>({});
+  const [usePipelineFromRunLabel, setUsePipelineFromRunLabel] = useState(
+    'Using pipeline from cloned run',
+  );
 
   // TODO(zijianjoy): If creating run from Experiment Page or RunList Page, there is no pipelineId/Version.
   const urlParser = new URLParser(props);
+  const originalRunId = urlParser.get(QUERY_PARAMS.cloneFromRun);
   const pipelineId = urlParser.get(QUERY_PARAMS.pipelineId);
   const pipelineVersionIdParam = urlParser.get(QUERY_PARAMS.pipelineVersionId);
+  const pipelineDetailsUrl = originalRunId
+    ? RoutePage.PIPELINE_DETAILS.replace(
+        ':' + RouteParams.pipelineId + '/version/:' + RouteParams.pipelineVersionId + '?',
+        '',
+      ) + urlParser.build({ [QUERY_PARAMS.fromRunId]: originalRunId })
+    : '';
+
+  // Retrieve Run Detail using RunID from backend
+  const { data: apiRun } = useQuery<ApiRunDetail, Error>(
+    ['ApiRun', originalRunId],
+    () => {
+      if (!originalRunId) {
+        throw new Error('No Run ID');
+      }
+      const originalRun = Apis.runServiceApi.getRun(originalRunId);
+      return originalRun;
+    },
+    { enabled: !!originalRunId, staleTime: Infinity },
+  );
+
+  let templateStringFromRunId;
+  if (apiRun) {
+    templateStringFromRunId = apiRun.run?.pipeline_spec?.pipeline_manifest;
+  }
+  const isTemplatePullSuccessFromRun = templateStringFromRunId !== undefined ? true : false;
+  const apiResourceRefFromRun = apiRun !== undefined ? apiRun?.run?.resource_references : undefined;
```

Comment:
`NewRunSwitcher` has already fetched this information, let's avoid refetching and try to pass in necessary information via `props` from `NewRunSwitcher` to here.